custom_components/tuya_local/devices/kogan_lx8_vacuum.yaml

```diff
@@ -0,0 +1,323 @@
+name: Kogan LX8 Robot Vacuum
+products:
+  - id: 1bgkj0zbhbw7fppy
+    name: Kogan LX8 Robot Vacuum
+primary_entity:
+  entity: vacuum
+  dps:
+    - id: 101
+      name: activate # clean_switch
+      type: boolean
+    - id: 102
+      name: pause # pause_switch
+      type: boolean
+    - id: 104
+      name: command # clean_mode
+      type: string
+      optional: true
+      mapping:
+        - value: pause
+        - value_redirect: pause
+        - dps_val: smart
+          value: smart
+        - dps_val: pose
+          value: clean_spot
+        - dps_val: zone
+          value: zone
+        - dps_val: backcharge
+          value: return_to_base
+    - id: 105
+      name: status # robot_state
+      type: string
+      mapping:
+        - dps_val: idle
+          value: standby
+        - dps_val: pointing\areaing
+          value: pointing_areaing
+        - dps_val: totaling
+          value: cleaning
+        - dps_val: sweep
+          value: sweep
+        - dps_val: mop
+          value: mop
+        - dps_val: fault
+          value: error
+        - dps_val: pause
+          value: paused
+        - dps_val: chargring
+          value: charging
+        - dps_val: tocharge
+          value: returning
+        - dps_val: fullcharge
+          value: charged
+        - dps_val: remotectl
+          value: remotectl
+    - id: 106
+      name: battery
```

Comment:
battery as an attribute on vacuums is deprecated since last July, and should be replaced by a battery sensor.
https://developers.home-assistant.io/blog/2023/07/10/vacuum-updates/

Reply:
copied the lx10 now